synapse/handlers/presence.py

```diff
@@ -1227,18 +1372,40 @@ async def set_state(
             return
 
         user_id = target_user.to_string()
+        now = self.clock.time_msec()
 
         prev_state = await self.current_state_for_user(user_id)
 
+        # Syncs do not override a previous presence of busy.
+        #
+        # XXX: This is *not* part of the MSC and is likely unneeded with multi-device support.
+        if prev_state.state == PresenceState.BUSY and is_sync:
+            presence = PresenceState.BUSY
```

Comment:
I *think* this is a hack for lack of multi-device support.

Note that it was added in #12213 and fixed in #14393. (#9644 added initial busy status stuff for reference.)

Reply:
Yes, I distinctly remember this being quite clunky and confusing without multi-device support and wanting to switch to a multi-device presence model, so yes, I think you're correct in that this is essentially a hack.

Reply:
Thanks! I suspect some clients that implement busy might depend on it though? We'll need to double check whenever this is stabilized.